[Issue-Docs] Update new-plugin-checklist.md (#11428)
microsoft/PowerToys: doc/devdocs/modules/launcher/new-plugin-checklist.md

- [ ] Microsoft plugin project name pattern: `Microsoft.PowerToys.Run.Plugin.{PluginName}`
- [ ] Community plugin project name pattern: `Community.PowerToys.Run.Plugin.{PluginName}`
- [ ] [`GlobalSuppressions.cs`](/src/codeAnalysis/GlobalSuppressions.cs) and [`StyleCop.json`](/src/codeAnalysis/StyleCop.json) have to be included in the plugin project so it follows PowerToys code guidelines

> - [ ] The plugin has to have `{PowerToys version}.0` version. It should be specified in the project file `<Version>$(Version).0</Version>`

Reviewer: ```suggestion
- [ ] The project file should import `Version.props` and specify `<Version>$(Version).0</Version>`
```
